src/registrar/views/domain.py

```diff
@@ -138,10 +138,19 @@ def get_initial(self):
         """The initial value for the form (which is a formset here)."""
         domain = self.get_object()
         nameservers = domain.nameservers
-        if nameservers is None:
-            return []
+        initial_data = []
 
-        return [{"server": name} for name, *ip in domain.nameservers]
+        if nameservers is not None:
+            # Add existing nameservers as initial data
+            initial_data.extend({"server": name} for name, *ip in nameservers)
+
+        # Ensure at least 3 fields, filled or empty
+        if not initial_data:
```

Comment:
was thinking this might be simpler with one clause, something like the following

        while len(initial_data) < 2:
            initial_data.append({})

Reply:
this comment above was meant to replace lines 148-151

Reply:
I agree with dave here

Reply:
So elegant